src/pages/settings/notification.vue

```diff
@@ -12,27 +12,73 @@
             </template>
           </SettingsListItem>
         </li>
-        <li>
+        <!-- <li>
           <SettingsListItem :is-clickable="false">
             {{ $t('settings_notification_tx_purchase') }}
             <template #accessory>
-              <CheckBox v-model="isPurchaseNotificationEnabled"/>
+              <CheckBox v-model="isPurchaseNotificationEnabled" />
             </template>
           </SettingsListItem>
-        </li>
+        </li> -->
       </ul>
     </section>
   </div>
 </template>
 
 <script>
+import { mapActions, mapGetters } from 'vuex';
+import { getUserNotification } from '~/util/api';
+
 export default {
   name: 'NotificationSettingsPage',
   data() {
     return {
+      isReady: false,
       isTransferNotificationEnabled: false,
       isPurchaseNotificationEnabled: false,
     };
   },
+  computed: {
+    ...mapGetters(['loginAddress', 'walletHasLoggedIn']),
+  },
+  watch: {
+    isTransferNotificationEnabled() {
+      if (this.isReady) {
+        this.handleToggle();
+      }
+    },
+  },
+  async mounted() {
+    if (!this.walletHasLoggedIn) {
+      try {
+        await this.signLogin();
+      } catch {
+        // No-op
+      }
+      if (!this.walletHasLoggedIn) {
+        return;
+      }
+    }
+    await this.fetchNotification();
+    this.isReady = true;
+  },
+  methods: {
+    ...mapActions(['signLogin']),
+    async fetchNotification() {
+      const { data } = await this.$api.get(
+        getUserNotification(this.loginAddress)
+      );
+      const { notification } = data;
+      const { transfer } = notification;
+      this.isTransferNotificationEnabled = transfer;
+    },
+    handleToggle() {
+      this.$api.post(getUserNotification(this.loginAddress), {
+        notification: {
+          transfer: this.isTransferNotificationEnabled,
+        },
+      });
+    },
+  },
```

Comment:
prefer generic function for this whole page, maybe `transfer` and `this.isTransferNotificationEnabled` as a input param when defining @change handler to call this function

Reply:
Seems `@input` event doesn't accept params, temporarily keep it.